File: Old_xml_and_scripts/Jim/plan/new-18.5.2013/planTournamentTurn.rb
### planTournamentTurn.rb ###
#	@author: A55Kickers		#
#############################

class PlanTournamentTurn < Plan
  
  def replan
    @plan.clear
    @agentInfo.whereIsBall();
    @agentInfo.isBallMine();
    @agentInfo.nearBall();
    @agentInfo.ballControlPosition();
    @agentInfo = Java::sk.fiit.jim.agent.AgentInfo.getInstance
    @agentModel = Java::sk.fiit.jim.agent.models.AgentModel.getInstance

    if EnvironmentModel.beamablePlayMode
    	@agentInfo.loguj('Beam');
    	@plan << Beam.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(-3, 0, 0.4))
    	@beamed = true
    elsif (me.on_ground? or me.is_lying_on_back? or me.is_lying_on_belly?)
    	@agentInfo.loguj('GetUp');
     	@plan << GetUp.new(Proc.new{me.on_ground?})
   	else
    	@agentInfo.loguj('Turn');
    	@plan << TurnToPosition.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(-10, 0, 0.4), Proc.new{not EnvironmentModel.beamablePlayMode})
    end
  end
end

File: Jim/src/sk/fiit/jim/garbage/plan/PlanTournamentTurn.java
package sk.fiit.jim.garbage.plan;

import sk.fiit.jim.LambdaCallable;
import sk.fiit.jim.agent.AgentInfo;
import sk.fiit.jim.agent.highskill.*;
import sk.fiit.jim.agent.models.*;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 * Rewrited PlanTournamentGetUpFromBelly from ruby
 * 
 * @author Homola
 */

public class PlanTournamentTurn extends Plan {

	public static PlanTournamentTurn getInstance() {
		return new PlanTournamentTurn();
	}
	
	public void replan() {     
		AgentInfo agentInfo = AgentInfo.getInstance();
        AgentModel agentModel = AgentModel.getInstance();
        Vector3D target_position = agentInfo.ballControlPosition();
		
        if (EnvironmentModel.beamablePlayMode())
        {
        	agentInfo.loguj("Beam");
        	Vector3D start_position = Vector3D.cartesian(-5, 1, 0.4);
			Highskill_Queue.add(new Beam(start_position));
        	beamed = true;
        }
        else if(agentModel.isOnGround() || agentModel.isLyingOnBack() || agentModel.isLyingOnBelly())
        {
        	agentInfo.loguj("GetUp");
			Highskill_Queue.add(new GetUp());
        }
        else
        {
        	agentInfo.loguj("Turn");
        	Highskill_Queue.add(new TurnToPosition(target_position,
        			new LambdaCallable() {
						public boolean call() {
							return ! turned_to_goal();
						}
					}
        	));
        }
	
	}
	
	
}
